extension/bootstrap.js

```diff
@@ -19,59 +20,136 @@ const REASONS = {
+      embeddedBrowser.setAttribute("flex", "1");
+
+      panel.appendChild(embeddedBrowser);
+      browserWindow.window.document.getElementById("mainPopupSet").appendChild(panel);
+    }
+    const burgerMenu = browserWindow.window.document.getElementById("PanelUI-menu-button");
+    // TODO What if there is no burger menu?
+    if (burgerMenu !== null) {
+      // only send the telemetry ping if we actually open the panel
+      studyUtils.telemetry({ treatment: "ask-to-add" });
+      panel.openPopup(burgerMenu, "bottomcenter topright", 0, 0, false, false);
+    }
+  } else if (shareButtonIsUseable(shareButton)) {
+    doorhangerDoNothingTreatment(browserWindow, browserWindow.shareButton);
+  }
+}
+
+function doorhangerAddToToolbarTreatment(browserWindow, shareButton) {
+  // TODO do not re-add to toolbar if user removed manually?
+
+  // check to see if the page will be shareable after adding the button to the toolbar
+  if (currentPageIsShareable(browserWindow) && !shareButtonIsUseable(shareButton)) {
+    studyUtils.telemetry({ treatment: "add-to-toolbar" });
+    CustomizableUI.addWidgetToArea("social-share-button", CustomizableUI.AREA_NAVBAR);
+    // need to get using browserWindow.shareButton because the shareButton argument
+    // was initialized before the button was added
+    doorhangerDoNothingTreatment(browserWindow, browserWindow.shareButton);
+  } else if (shareButtonIsUseable(shareButton)) {
+    doorhangerDoNothingTreatment(browserWindow, browserWindow.shareButton);
+  }
 }
 
 // define treatments as STRING: fn(browserWindow, shareButton)
 const TREATMENTS = {
-  doorhanger: doorhangerTreatment,
-  highlight: highlightTreatment,
+  highlight:              highlightTreatment,
+  doorhangerDoNothing:    doorhangerDoNothingTreatment,
+  doorhangerAskToAdd:     doorhangerAskToAddTreatment,
+  doorhangerAddToToolbar: doorhangerAddToToolbarTreatment,
 };
 
 async function chooseVariation() {
-  let variation;
   const sample = studyUtils.sample;
-
-  if (config.study.variation) {
-    variation = config.study.variation;
-  } else {
-    // this is the standard arm choosing method
-    const clientId = await studyUtils.getTelemetryId();
-    const hashFraction = await sample.hashFraction(config.study.studyName + clientId);
-    variation = sample.chooseWeighted(config.study.weightedVariations, hashFraction);
+  // this is the standard arm choosing method
+  const clientId = await studyUtils.getTelemetryId();
+  const hashFraction = await sample.hashFraction(config.study.studyName + clientId);
+  let variation = sample.chooseWeighted(config.study.weightedVariations, hashFraction);
+
+  // if pref has a user-set value, use this instead
+  if (Preferences.isSet(TREATMENT_OVERRIDE_PREF)) {
+    variation = { name: Preferences.get(TREATMENT_OVERRIDE_PREF, variation),
+      weight: 1 };
+    Preferences.reset(TREATMENT_OVERRIDE_PREF); // reset to default branch value to clear
```

Comment:
I think that this shouldn't actually get cleared. It is ok to clear the counter pref, since that is really internal state. But this is user choice (well, tester choice) that I wouldn't expect to get poofed just because the add-on got uninstalled. It also isn't a matter of cleaning up after the extension, since this pref will only be set manually.